src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -802,13 +1022,13 @@ private static void RenameCodeInterfaceParamsInSerializers(CodeFunction codeFunc
     {
         if (codeFunction.OriginalLocalMethod.Parameters.FirstOrDefault(static x => x.Type is CodeType codeType && codeType.TypeDefinition is CodeInterface) is CodeParameter param && param.Type is CodeType codeType && codeType.TypeDefinition is CodeInterface paramInterface)
         {
-            param.Name = ReturnFinalInterfaceName(paramInterface);
+            param.Name = GetFinalInterfaceName(paramInterface);
         }
     }
 
-    private static string ReturnFinalInterfaceName(CodeInterface codeInterface)
+    private static string GetFinalInterfaceName(CodeInterface codeInterface)
     {
-        return codeInterface.OriginalClass.Name.ToFirstCharacterUpperCase();
+        return codeInterface.OriginalClass?.Name.ToFirstCharacterUpperCase() ?? throw new InvalidOperationException($"The refiner was unable to find the original class for {codeInterface.Name}");
```

Comment:
`OriginalClass` can't be null. It is a required property. 
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required